### DISCUSSION TITLE
params passed to a test

### DISCUSSION THREAD

**Discussion by caridm** [pytest-dev/pytest#11093] [General]
Hi, I am moving our testing infrastructure from python 2.7 to python 3.6 (I know we are behind in versions, but this is what I have to work with) and am using pytest 7.0.1.   In the older version of pytest that we used the test, when called, got the parameters that were passed as a dictionary.  When using version 7.0.1 the parameters are now inside a pytest.mark.   The code that sets up the parameters via pytest_generate_tests does the following:

            params.append(pytest.mark.timeout(timeout_sec,dict(iteration=int(force_iteration), flags=flags,
                                                               test_level=CYC_ALL_TEST_LEVELS[run_level],
                                                               level=run_level, name=metafunc.function.__name__, module_name=module_name)))
            metafunc.parametrize('params', params)

So I understand why they are inside a pytest.mark, however it seems as if in the older version of pytest, the parameters i.e. the dictionary were removed from the mark and passed to my test, in the newer version I am getting the pytest.mark.

I have been going thru the documentation to see if there is something to tell me why this change occurred and what I can do to keep that behavior so that I don't have to change the 5000+ tests that may be affected.

Thanks for any help you can give.

**Comment by RonnyPfannschmidt** [comment]
The transformation of botched # # marks to parameter sets has been replaced bye pytest.param since ages, off hand im not sure when it was added and when the removal hit
